lib/features/children/family_history/models/allowance.dart

```diff
@@ -24,6 +28,22 @@ class Allowance extends HistoryItem {
       type: HistoryTypes.values.firstWhere(
         (element) => element.value == map['donationType'],
       ),
+      status: AllowanceStatus.values.firstWhere(
```

Comment:
I would suggest to encapsulate such search and put it to the AllowanceStatus enum. Also perhaps add error handling (when filed is empty or has another value)?